app/src/main/java/org/gdg/frisbee/android/utils/PlusUtils.java

```diff
@@ -1,28 +1,36 @@
 package org.gdg.frisbee.android.utils;
 
-import android.net.Uri;
-import android.support.annotation.NonNull;
+import android.content.Context;
 import android.support.annotation.Nullable;
 
+import com.google.android.gms.auth.api.Auth;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInResult;
 import com.google.android.gms.common.api.GoogleApiClient;
-import com.google.android.gms.plus.Plus;
-import com.google.android.gms.plus.model.people.Person;
+import com.google.android.gms.common.api.OptionalPendingResult;
+
+import org.gdg.frisbee.android.app.GoogleApiClientFactory;
 
 public class PlusUtils {
 
-    private PlusUtils() {
+    @Nullable
+    public static String getCurrentPersonId(Context context) {
+        GoogleSignInAccount account = getCurrentAccount(context);
+        return account != null ? account.getId() : null;
     }
 
     @Nullable
-    public static String getCurrentPersonId(GoogleApiClient apiClient) {
-        Person plusPerson = null;
-        if (apiClient.isConnected() && apiClient.hasConnectedApi(Plus.API)) {
-            plusPerson = Plus.PeopleApi.getCurrentPerson(apiClient);
+    public static GoogleSignInAccount getCurrentAccount(Context context) {
+        GoogleApiClient client = GoogleApiClientFactory.createForSignIn(context);
+
+        OptionalPendingResult<GoogleSignInResult> result = Auth.GoogleSignInApi.silentSignIn(client);
+        if (!result.isDone()) {
```

Comment:
When isDone is true, it means that everything is ready and locally saved. So it should be instant. 

And the beat thing is, it will be false automatically when user goes into setting in web and revokes permissions for our app.